xla/service/collective_ops_utils.cc

```diff
@@ -133,6 +133,31 @@ absl::StatusOr<std::vector<int>> GetParticipatingIDs(
                           group->replica_ids().end());
 }
 
+// Returns the group formation mode of instr, assuming that instr is, or is
+// dervied from, an HloAllGatherInstruction, HloAllReduceInstructionBase,
+// HloCollectiveInstruction or HloCollectivePermuteInstruction.
```

Comment:
No need to duplicate comment from header. Also no need to explicitly call out all-gather and all-reduce since they already derive from HloCollectiveInstruction. But maybe it's better to list the HLO collective names instead of classes, e.g. "all-gather, all-reduce", in which case you should list them all.